src/Core/OrganizationFeatures/OrganizationSponsorships/FamiliesForEnterprise/SelfHosted/SelfHostedSyncSponsorshipsCommand.cs

```diff
@@ -0,0 +1,129 @@
+    {
+        private readonly GlobalSettings _globalSettings;
+        private readonly IOrganizationSponsorshipRepository _organizationSponsorshipRepository;
+        private readonly IOrganizationUserRepository _organizationUserRepository;
+        private readonly IOrganizationConnectionRepository _organizationConnectionRepository;
+        private readonly ILicensingService _licensingService;
+
+        public SelfHostedSyncSponsorshipsCommand(
+        IOrganizationSponsorshipRepository organizationSponsorshipRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        IOrganizationConnectionRepository organizationConnectionRepository,
+        ILicensingService licensingService,
+        GlobalSettings globalSettings,
+        ILogger<SelfHostedSyncSponsorshipsCommand> logger)
+        : base(
+            globalSettings.BaseServiceUri.InternalVault,
+            globalSettings.BaseServiceUri.InternalIdentity,
+            "api.installation",
+            globalSettings.Installation.Id.ToString(),
+            globalSettings.Installation.Key,
+            logger)
+        {
+            _globalSettings = globalSettings;
+            _organizationUserRepository = organizationUserRepository;
+            _organizationSponsorshipRepository = organizationSponsorshipRepository;
+            _organizationConnectionRepository = organizationConnectionRepository;
+            _licensingService = licensingService;
+        }
+
+        public async Task SyncOrganization(Guid organizationId, OrganizationConnection billingSyncKey)
+        {
+            if (!_globalSettings.EnableCloudCommunication)
+            {
+                _logger.LogInformation("Failed to sync instance with cloud - Cloud communication is disabled in global settings");
+                return;
+            }
+            if (!billingSyncKey.Enabled)
+            {
+                throw new BadRequestException($"Billing Sync Key disabled for organization {organizationId}");
+            }
+            if (string.IsNullOrWhiteSpace(billingSyncKey.Config))
+            {
+                throw new BadRequestException($"No Billing Sync Key known for organization {organizationId}");
+            }
+
+            var cloudOrganizationId = (await _licensingService.ReadOrganizationLicenseAsync(organizationId)).Id;
+            var organizationSponsorships = await _organizationSponsorshipRepository.GetManyBySponsoringOrganizationAsync(organizationId);
+            var syncedSponsorships = new List<OrganizationSponsorshipData>();
+
+            foreach (var orgSponsorshipsBatch in CoreHelpers.Batch(organizationSponsorships, 1000))
+            {
+                var response = await SendAsync<OrganizationSponsorshipSyncRequestModel, OrganizationSponsorshipSyncResponseModel>(HttpMethod.Post, "organizationSponsorships/sync", new OrganizationSponsorshipSyncRequestModel
+                {
+                    BillingSyncKey = billingSyncKey.Config,
```

Comment:
I believe the intention is that config holds some json data, not that it is the key directly. @justindbaur, do you have an update on the config object here and how to use it?